### ISSUE TITLE
Crash when downloading video from Twitter

### ISSUE THREAD

**Issue by EmmanuelMess** [TeamAmaze/AmazeFileManager#928]
Share [this](https://twitter.com/DavidSchenet/status/952746916495978501) with Amaze, click the FAB and crash.

```
? E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #4
                    Process: com.amaze.filemanager, PID: 30639
                    java.lang.RuntimeException: An error occurred while executing doInBackground()
                        at android.os.AsyncTask$3.done(AsyncTask.java:309)
                        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
                        at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
                        at java.util.concurrent.FutureTask.run(FutureTask.java:242)
                        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
                        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
                        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
                        at java.lang.Thread.run(Thread.java:818)
                     Caused by: java.lang.NullPointerException: uri
                        at com.android.internal.util.Preconditions.checkNotNull(Preconditions.java:60)
                        at android.content.ContentResolver.openInputStream(ContentResolver.java:645)
                        at com.amaze.filemanager.filesystem.FileUtil$1.doInBackground(FileUtil.java:188)
                        at com.amaze.filemanager.utils.application.AppConfig$1.doInBackground(AppConfig.java:106)
                        at com.amaze.filemanager.utils.application.AppConfig$1.doInBackground(AppConfig.java:90)
                        at android.os.AsyncTask$2.call(AsyncTask.java:295)
                        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234) 
                        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) 
                        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) 
                        at java.lang.Thread.run(Thread.java:818) 
```

**Comment by EmmanuelMess** [comment]
I thought it would generalize to all videos.

**Comment by VishalNehra** [comment]
Seems like I would need to do more testing with this option.

**Comment by TranceLove** [comment]
By the way, Twitter's `ACTION_SEND` intent does not contain `Intent.EXTRA_STREAM` attribute in the intent's `Bundle`.

Therefore a quick fix would be adding null check if `Intent.EXTRA_STREAM` correctly returned an `Uri`. Let me work on this.

**Comment by jjakob** [comment]
Also happens when sharing from BinaryEye:

Steps to reproduce: share text from BinaryEye, select Amaze "Save as...", select folder to save to and save.

```
08-05 10:04:11.174  1171  1171 W WindowManager: Attempted to remove non-existing token: android.os.Binder@9b65e7c
08-05 10:04:13.713  1171  4966 I PackageManager: Setting last chosen activity com.amaze.filemanager/.activities.MainActivity for user 0:
08-05 10:04:13.713  1171  4966 I PackageManager:   Action: "android.intent.action.SEND"
08-05 10:04:13.713  1171  4966 I PackageManager:   Category: "android.intent.category.DEFAULT"
08-05 10:04:13.713  1171  4966 I PackageManager:   Type: "text/plain"
08-05 10:04:13.713  1171  4966 I PackageManager:   AutoVerify=false
08-05 10:04:13.717  1171  6770 I ActivityManager: START u0 {act=android.intent.action.SEND typ=text/plain flg=0x3000001 cmp=com.amaze.filemanager/.activities.MainActivity clip={text/plain T:0000011013213453} (has extras)} from uid 10143 on display 0
08-05 10:04:15.114   604   604 I MSM-irqbalance: Decided to move IRQ200 from CPU0 to CPU3
08-05 10:04:15.695 10609 10643 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #2
08-05 10:04:15.695 10609 10643 E AndroidRuntime: Process: com.amaze.filemanager, PID: 10609
08-05 10:04:15.695 10609 10643 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at android.os.AsyncTask$3.done(AsyncTask.java:325)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at java.util.concurrent.FutureTask.run(FutureTask.java:242)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:761)
08-05 10:04:15.695 10609 10643 E AndroidRuntime: Caused by: java.lang.NullPointerException: uri
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at com.android.internal.util.Preconditions.checkNotNull(Preconditions.java:111)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at android.content.ContentResolver.openInputStream(ContentResolver.java:689)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at com.amaze.filemanager.filesystem.FileUtil$1.doInBackground(FileUtil.java:209)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at com.amaze.filemanager.filesystem.FileUtil$1.doInBackground(FileUtil.java:198)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at com.amaze.filemanager.utils.application.AppConfig$1.doInBackground(AppConfig.java:122)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at android.os.AsyncTask$2.call(AsyncTask.java:305)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
08-05 10:04:15.695 10609 10643 E AndroidRuntime:        ... 3 more```

**Comment by VishalNehra** [comment]
Should be fixed with 3.5.0
Please test and reopen if required.
